api/lib/searchOperators.js

```diff
@@ -0,0 +1,121 @@
+module.exports = {
+  /**
+   * Convert ang search operators to database query
+   *
+   * @since 2.1.2
+   * @param {string} single page no, or page nos with operators
+   * @returns {string} Returns database WHERE clauses
+   * @example
+   *
+   * angToQuery: <1000+105+900-905
+   * // => {
+   *       q: 'v.PageNo < ? AND (v.PageNo BETWEEN ? AND ? OR v.PageNo IN (?))',
+   *       parameters: [ '1000', '900', '905', '105' ],
+   *       least: 0,
+   *       most: '905',
+   *       totalPages: null
+   *       }
+   */
+  angToQuery: PageNo => {
+    const hasOperatorsRegEx = /[<>+-]/gm;
+
+    if (!PageNo.match(hasOperatorsRegEx)) {
+      return {
+        q: 'v.PageNo = ?',
+        parameters: [PageNo],
+        least: PageNo,
+        most: PageNo,
+        totalPages: 1,
+      };
+    }
+
+    const qOR = [];
+    const qAND = [];
+    const qIn = [];
+    const parameters = [];
+    let least = 0;
+    let most = 0;
+
+    const ltgtRegEx = /[<>][0-9]+/gm;
+
+    const ltgt = PageNo.match(ltgtRegEx);
+    if (ltgt) {
+      let loopnum = 0;
+      ltgt.forEach(num => {
+        if (!num.match(/-$/)) {
```

Comment:
Is this match trying to see if there is a `-` at the end of `num`? If so, I don't think this check is necessary? Each `num` is supposed to match `/[<>][0-9]+` which means the ending of each match will be a numeric digit.. so we'd never find a `-` at the end?

Reply:
so the reason for the double set of regex is, we want it to match +105 (add page 105) but not +105-110 (add pages 105-110) because the latter is handled by the regex for ranges, so we match +105 or +105- and then we discard the ones that end in -